File: src/app/Models/Donation.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Donation extends Model
{
    protected $fillable = [
        'invoice_id',
        'campaign_id',
        'user_id',
        'donor_name',
        'amount',
    ];

    public function invoice(): BelongsTo
    {
        return $this->belongsTo(Invoice::class);
    }

    public function campaign(): BelongsTo
    {
        return $this->belongsTo(Campaign::class);
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}

File: backend/internal/model/donation.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type Donation struct {
	ID         uuid.UUID  `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	InvoiceID  uuid.UUID  `gorm:"type:uuid;not null;index" json:"invoice_id"`
	CampaignID uuid.UUID  `gorm:"type:uuid;not null;index" json:"campaign_id"`
	UserID     *uuid.UUID `gorm:"type:uuid;index" json:"user_id"`
	DonorName  string     `gorm:"size:255" json:"donor_name"`
	Amount     int64      `gorm:"not null" json:"amount"`
	CreatedAt  time.Time  `json:"created_at"`
	UpdatedAt  time.Time  `json:"updated_at"`

	Invoice  Invoice   `gorm:"foreignKey:InvoiceID;constraint:OnDelete:CASCADE" json:"invoice,omitempty"`
	Campaign Campaign  `gorm:"foreignKey:CampaignID;constraint:OnDelete:CASCADE" json:"campaign,omitempty"`
	User     *User     `gorm:"foreignKey:UserID;constraint:OnDelete:SET NULL" json:"user,omitempty"`
}
